### PR TITLE
fix: reply hasHeader behavior

### PR DESCRIPTION
**Pull request by zekth**
fix: https://github.com/fastify/fastify/issues/3030

@matteo said 

> it should be based on the same logic. We can avoid generating the wrapper object.

src: https://github.com/fastify/fastify/issues/3030#issuecomment-825156516

Issue is in a case of a proxyfied reply the nodeJs raw reply still has its own headers so this it's still needed:
https://github.com/fastify/fastify/blob/28febddefce64d91e2618fd14001211130a067ad/lib/reply.js#L204-L209

#### Benchmarks
before:
```
[1] ┌─────────┬──────┬───────┬───────┬───────┬──────────┬─────────┬────────┐
[1] │ Stat    │ 2.5% │ 50%   │ 97.5% │ 99%   │ Avg      │ Stdev   │ Max    │
[1] ├─────────┼──────┼───────┼───────┼───────┼──────────┼─────────┼────────┤
[1] │ Latency │ 7 ms │ 17 ms │ 30 ms │ 56 ms │ 18.04 ms │ 8.17 ms │ 158 ms │
[1] └─────────┴──────┴───────┴───────┴───────┴──────────┴─────────┴────────┘
[1] ┌───────────┬─────────┬─────────┬─────────┬───────┬─────────┬─────────┬─────────┐
[1] │ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5% │ Avg     │ Stdev   │ Min     │
[1] ├───────────┼─────────┼─────────┼─────────┼───────┼─────────┼─────────┼─────────┤
[1] │ Req/Sec   │ 40959   │ 40959   │ 57439   │ 58559 │ 54275.2 │ 6729.92 │ 40957   │
[1] ├───────────┼─────────┼─────────┼─────────┼───────┼─────────┼─────────┼─────────┤
[1] │ Bytes/Sec │ 7.66 MB │ 7.66 MB │ 10.7 MB │ 11 MB │ 10.1 MB │ 1.26 MB │ 7.66 MB │
[1] └───────────┴─────────┴─────────┴─────────┴───────┴─────────┴─────────┴─────────┘
```
after:
```
[1]
[1] ┌─────────┬──────┬───────┬───────┬───────┬──────────┬─────────┬────────┐
[1] │ Stat    │ 2.5% │ 50%   │ 97.5% │ 99%   │ Avg      │ Stdev   │ Max    │
[1] ├─────────┼──────┼───────┼───────┼───────┼──────────┼─────────┼────────┤
[1] │ Latency │ 6 ms │ 17 ms │ 28 ms │ 58 ms │ 17.55 ms │ 8.66 ms │ 131 ms │
[1] └─────────┴──────┴───────┴───────┴───────┴──────────┴─────────┴────────┘
[1] ┌───────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┬─────────┐
[1] │ Stat      │ 1%      │ 2.5%    │ 50%     │ 97.5%   │ Avg     │ Stdev   │ Min     │
[1] ├───────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
[1] │ Req/Sec   │ 41631   │ 41631   │ 59423   │ 60351   │ 55689.6 │ 7093.95 │ 41620   │
[1] ├───────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┤
[1] │ Bytes/Sec │ 7.79 MB │ 7.79 MB │ 11.1 MB │ 11.3 MB │ 10.4 MB │ 1.33 MB │ 7.78 MB │
[1] └───────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┘
[1]
```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/reply.js [hunk 1]
```diff
@@ -209,7 +209,10 @@ Reply.prototype.getHeaders = function () {
 }
 
 Reply.prototype.hasHeader = function (key) {
-  return this[kReplyHeaders][key.toLowerCase()] !== undefined
+  if (this[kReplyHeaders][key.toLowerCase()] !== undefined) {
```
Comment (mcollina) [inline lib/reply.js:212]:
```suggestion
  key = key.toLowerCase()
  if (this[kReplyHeaders][key] !== undefined) {
```

**Review by mcollina — COMMENTED** [review]
Can you run the benchmarks as well?

**Comment by zekth** [comment]
Benchmarks added.

Side note, we can also use the branch free approach like:
```js
Reply.prototype.hasHeader = function (key) {
  key = key.toLowerCase()
  return this[kReplyHeaders][key] !== undefined || this.raw.hasHeader(key)
}
```

**Review by jsumners — COMMENTED** [review]
Can we add a test to cover this?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**